server/src/main/java/org/apache/kafka/server/metrics/ClientMetricsConfigs.java

```diff
@@ -131,9 +131,11 @@ private static void validateProperties(Properties properties) {
             }
         });
 
+        ClientMetricsConfigs configs = new ClientMetricsConfigs(properties);
```

Comment:
What happens if there are configs having no default value?

Reply:
Hi @chia7712 , thanks for review.
The situation can't happen because `Line 120` checks the properties which is defined at `Line 102` and they have default value except `PUSH_INTERVAL_MS `.

Reply:
I am not convinced instantiating `ClientMetricsConfigs(properties)` in the same class itself. The idea with validate is to create `ClientMetricsConfigs` instance only after post validating the configs are correct.

I understand we want to thrown `invalid request exception` when configs are not correct and as per the changes I am assuming that the unknown server occurs when the int config is specified as non-int, correct? If that's the only scenario then we can handle directly. Am I reading something wrong here?

Reply:
> I am not convinced instantiating `ClientMetricsConfigs(properties)` in the validate method. The idea with validate is to create `ClientMetricsConfigs` instance only after post validating the configs are correct.
> 
> I understand we want to thrown `invalid request exception` when configs are not correct and as per the changes I am assuming that the unknown server occurs when the int config is specified as non-int, correct? If that's the only scenario then we can handle directly. Am I reading something wrong here?

Yes, you are right and I updated this part by utilize `ConfigDef.parseType`. 
PTAL.